libraries/apollo-ast/src/commonMain/kotlin/com/apollographql/apollo3/ast/Issue.kt

```diff
@@ -59,6 +59,17 @@ class IncompatibleDirectiveDefinition(
   override val message = "Unexpected '@$directiveName' directive definition. Expecting '$expectedDefinition'."
 }
 
+/**
+ * The enum definition is inconsistent with the expected one.
+ */
+class IncompatibleEnumDefinition(
+    enumName: String,
+    expectedDefinition: String,
+    override val sourceLocation: SourceLocation?,
+) : GraphQLValidationIssue {
+  override val message = "Unexpected '@$enumName' enum definition. Expecting '$expectedDefinition'."
+}
```

Comment:
Should we go for a generic `IncompatibleDefinition` instead? If we want to have the comparison code generic at some point?